lib/utils/index.js

```diff
@@ -1058,17 +1058,31 @@ function unsigned_url_prefix(
   return prefix;
 }
 
-function base_api_url(path = [], options = {}) {
-  let cloudinary = ensureOption(options, "upload_prefix", UPLOAD_PREFIX);
-  let cloud_name = ensureOption(options, "cloud_name");
-  let encode_path = unencoded_path => encodeURIComponent(unencoded_path).replace("'", '%27');
-  let encoded_path = Array.isArray(path) ? path.map(encode_path) : encode_path(path);
-  return [cloudinary, "v1_1", cloud_name].concat(encoded_path).join("/");
+function base_api_url_v1() {
+  return base_api_url('v1_1');
+}
+
+function base_api_url_v2() {
+  return base_api_url('v2');
+}
+
+function base_api_url(api_version) {
+  if (!api_version || api_version.length === 0) {
+    throw new Error('api_version needs to be a non-empty string');
+  }
+
+  return (path = [], options = []) => {
```

Comment:
This looks kinda weird to me. Is this idiomatic to node?

Reply:
It's a very common pattern called function currying: https://javascript.info/currying-partials